src/lib/viewers/controls/media/MediaSettingsControls/MediaSettingsFlyout.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Menu } from './MediaSettingsMenu';
+import './MediaSettingsFlyout.scss';
+
+export type Props = {
+    children: React.ReactNode;
+    className?: string;
+    menu: Menu;
+    menuSelectors: Record<string, string>;
+};
+
+const SETTINGS_MENU_PADDING = 18;
+const SETTINGS_MENU_PADDING_SCROLL = 32;
+const SETTINGS_MENU_MAX_HEIGHT = 210;
+
+export default function MediaSettingsFlyout({ children, className, menu, menuSelectors }: Props): JSX.Element | null {
+    const [dimension, setDimension] = React.useState({});
+    const flyoutRef = React.useRef<HTMLDivElement>(null);
+
+    const handleTransitionEnd = (): void => {
+        const { current: flyoutEl } = flyoutRef;
+
+        if (flyoutEl) {
+            flyoutEl.classList.remove('bp-MediaSettingsFlyout-in-transition');
+        }
+    };
+
+    React.useEffect(() => {
+        const { current: flyoutEl } = flyoutRef;
+
+        if (flyoutEl) {
+            flyoutEl.addEventListener('transitionend', handleTransitionEnd);
```

Comment:
Can this live on the `<div>` as `onTransitionEnd`? If not, can we add a brief comment in code as to why?